### ISSUE TITLE
win: cannot read the battery info properly

### ISSUE THREAD

**Issue by babojamo** [sebhildebrandt/systeminformation#641]
**Describe the bug**

I am not getting accurate battery information. AC is always connected and the percentage is always 0
I inspect the battery.js and console the return from WMI

```
PSComputerName              : DESKTOP-HBA5R8G
__GENUS                     : 2
__CLASS                     : Win32_Battery
__SUPERCLASS                : CIM_Battery
__DYNASTY                   : CIM_ManagedSystemElement
__RELPATH                   : Win32_Battery.DeviceID=" 4390BYDDELL M4GWP11"
__PROPERTY_COUNT            : 33
__DERIVATION                : {CIM_Battery, CIM_LogicalDevice, CIM_LogicalElement, CIM_ManagedSystemElement}
__SERVER                    : DESKTOP-HBA5R8G
__NAMESPACE                 : root\cimv2
__PATH                      : \\DESKTOP-HBA5R8G\root\cimv2:Win32_Battery.DeviceID=" 4390BYDDELL M4GWP11"
Availability                : 2
BatteryRechargeTime         : 
BatteryStatus               : 2
Caption                     : Internal Battery
Chemistry                   : 2
ConfigManagerErrorCode      : 
ConfigManagerUserConfig     : 
CreationClassName           : Win32_Battery
Description                 : Internal Battery
DesignCapacity              : 
DesignVoltage               : 13165
DeviceID                    :  4390BYDDELL M4GWP11
ErrorCleared                : 
ErrorDescription            : 
EstimatedChargeRemaining    : 98
EstimatedRunTime            : 71582788
ExpectedBatteryLife         : 
ExpectedLife                : 
FullChargeCapacity          : 
InstallDate                 : 
LastErrorCode               : 
MaxRechargeTime             : 
Name                        : DELL M4GWP11
PNPDeviceID                 : 
PowerManagementCapabilities : {1}
PowerManagementSupported    : False
SmartBatteryVersion         : 
Status                      : OK
StatusInfo                  : 
SystemCreationClassName     : Win32_ComputerSystem
SystemName                  : DESKTOP-HBA5R8G
TimeOnBattery               : 
TimeToFullCharge            : 
Scope                       : System.Management.ManagementScope
Path                        : \\DESKTOP-HBA5R8G\root\cimv2:Win32_Battery.DeviceID=" 4390BYDDELL M4GWP11"
Options                     : System.Management.ObjectGetOptions
ClassPath                   : \\DESKTOP-HBA5R8G\root\cimv2:Win32_Battery
Properties                  : {Availability, BatteryRechargeTime, BatteryStatus, Caption...}
SystemProperties            : {__GENUS, __CLASS, __SUPERCLASS, __DYNASTY...}
Qualifiers                  : {dynamic, Locale, provider, UUID}
Site                        : 
Container                   : 

```

**To Reproduce**
read from si.getAllData()
```
  async getSystemInfo() {
    await siLib.getAllData()
      .then(data => {
        console.log(data.battery);
      })
      .catch(error => {
        throw error;
      });
  }
```
Console Log
```
{
    "hasBattery": true,
    "maxCapacity": 43138,
    "designedCapacity": 51004,
    "voltage": 0,
    "capacityUnit": "mWh",
    "percent": 0,
    "currentCapacity": 0,
    "isCharging": true,
    "timeRemaining": null,
    "acConnected": true,
    "model": "",
    "type": "",
    "manufacturer": "",
    "serial": ""
}
```
 - OS: Microsoft Windows 10 Home Single Language
 - Hardware Dell G5 SE

**Comment by si458** [comment]
@babojamo it could be a battery issue?
ive checked the powershell commands on my Parellels Win10 VM on Mac and it returns battery information no problem?
although mac says the batterys are 100% and windows says its say '96% and plugged in' so who knows with windows haha
```
PS C:\Users\Simon> Get-WmiObject Win32_Battery | fl *


PSComputerName              : SIWIN10MAC
__GENUS                     : 2
__CLASS                     : Win32_Battery
__SUPERCLASS                : CIM_Battery
__DYNASTY                   : CIM_ManagedSystemElement
__RELPATH                   : Win32_Battery.DeviceID="SMPbq20z45"
__PROPERTY_COUNT            : 33
__DERIVATION                : {CIM_Battery, CIM_LogicalDevice, CIM_LogicalElement, CIM_ManagedSystemElement}
__SERVER                    : SIWIN10MAC
__NAMESPACE                 : root\cimv2
__PATH                      : \\SIWIN10MAC\root\cimv2:Win32_Battery.DeviceID="SMPbq20z45"
Availability                : 2
BatteryRechargeTime         :
BatteryStatus               : 2
Caption                     : Internal Battery
Chemistry                   : 2
ConfigManagerErrorCode      :
ConfigManagerUserConfig     :
CreationClassName           : Win32_Battery
Description                 : Internal Battery
DesignCapacity              :
DesignVoltage               : 12797
DeviceID                    : SMPbq20z45
ErrorCleared                :
ErrorDescription            :
EstimatedChargeRemaining    : 96
EstimatedRunTime            : 71582788
ExpectedBatteryLife         :
ExpectedLife                :
FullChargeCapacity          :
InstallDate                 :
LastErrorCode               :
MaxRechargeTime             :
Name                        : bq20z45
PNPDeviceID                 :
PowerManagementCapabilities : {1}
PowerManagementSupported    : False
SmartBatteryVersion         :
Status                      : OK
StatusInfo                  :
SystemCreationClassName     : Win32_ComputerSystem
SystemName                  : SIWIN10MAC
TimeOnBattery               :
TimeToFullCharge            :
Scope                       : System.Management.ManagementScope
Path                        : \\SIWIN10MAC\root\cimv2:Win32_Battery.DeviceID="SMPbq20z45"
Options                     : System.Management.ObjectGetOptions
ClassPath                   : \\SIWIN10MAC\root\cimv2:Win32_Battery
Properties                  : {Availability, BatteryRechargeTime, BatteryStatus, Caption...}
SystemProperties            : {__GENUS, __CLASS, __SUPERCLASS, __DYNASTY...}
Qualifiers                  : {dynamic, Locale, provider, UUID}
Site                        :
Container                   :



PS C:\Users\Simon> (Get-WmiObject -Class BatteryStaticData -Namespace ROOT/WMI).DesignedCapacity
100000
PS C:\Users\Simon> (Get-WmiObject -Class BatteryFullChargedCapacity -Namespace ROOT/WMI).FullChargedCapacity
100000
PS C:\Users\Simon>
```

**Comment by si458** [comment]
IGNORE ME im shatting out my bum this morning,
same issue here when running with nodejs (forgot to check that first instead of powershell)
although the information thats returned is the same as my post above https://github.com/sebhildebrandt/systeminformation/issues/641#issuecomment-1014379338
`si.get({battery:'*'}).then((data)=>{ console.log(data); });`
```
{
  battery: {
    hasBattery: true,
    maxCapacity: 0,
    designedCapacity: 0,
    voltage: 0,
    capacityUnit: 'mWh',
    percent: 0,
    currentCapacity: 0,
    isCharging: true,
    timeRemaining: null,
    acConnected: true,
    model: '',
    type: '',
    manufacturer: '',
    serial: ''
  }
}
```

**Comment by sebhildebrandt** [comment]
@babojamo @si458 seems that two of the three powershell commands return data (inside nodes) but the third one not. Need to check this here locally.

**Comment by si458** [comment]
@babojamo please can you try applying my patch #642 ? i believe this fixes the issue 👍

**Comment by sebhildebrandt** [comment]
@babojamo if you can confirm that the PR #642 fixes it also on your side, I will immediately publish a new version. @si458 Thank you for your support!!!

**Comment by si458** [comment]
@sebhildebrandt just pushed new commit to PR
new commit fixes the `designedCapacity` and `fullChargeCapacity` for use with `powershellStart()`

**Comment by babojamo** [comment]
@si458 @sebhildebrandt 
Cheers! That fixed the issue, it can now read the battery information!

Thank you so much!

**Comment by sebhildebrandt** [comment]
@babojamo @si458 just published a new version 5.10.4 with the PR you provided. Can you test it on your side? Thank you!

**Comment by si458** [comment]
@sebhildebrandt i can indeed confirm the battery works as expected, both with `powerShellStart()` and without `powerShellStart()`
